code/linalg.c

```diff
@@ -183,6 +183,21 @@ static mp_obj_t linalg_dot(mp_obj_t _m1, mp_obj_t _m2) {
 
 MP_DEFINE_CONST_FUN_OBJ_2(linalg_dot_obj, linalg_dot);
 
+static mp_obj_t linalg_norm(mp_obj_t _x) {
+    if (!MP_OBJ_IS_TYPE(_x, &ulab_ndarray_type)) {
```

Comment:
If it is not very expensive, I would allow generic iterables. That would mean that the `for` loop has to be implemented once more. The rest is OK.